### ISSUE TITLE
[Feature Request] Add success condition `all but first`

### ISSUE THREAD

**Issue by woody34** [open-cli-tools/concurrently#280]
## Description

I would like to add a third success condition that would allow you to spin up a service, complete a list of tasks in parallel, capture their exit status. Assuming all tasks exited successfully, exit application with success status.

## Use case

I currently use concurrently to spin up storybook and complete cypress tests against a component library. I would like to be able to also perform jest visual regression tests.

## Proposed Use Case

```
const concurrently = require('concurrently');
const waitForStorybook = 'npx wait-on http://localhost:6006/';
concurrently(
	[
		{ name: 'Storybook', command: 'npm:sb', prefixColor: 'bgMagenta.bold' },
		{ name: 'Cypress', command: `${waitForStorybook} && npm run test:cypress:run`, prefixColor: 'bgGreen.bold' },
		{ name: 'Jest', command: `${waitForStorybook} && npm run test:unit`, prefixColor: 'bgBlue.bold' },
	],
	{
		prefix: 'name',
		successCondition: 'all-but-first',
	},
)
	.catch(console.error);
```

I am currently nesting concurrently calls to achieve this behavior and it doesn't provide the best logs.

## Proposed Acceptance Criteria

1. run 3 tasks concurrently
2. once task 2 and 3 complete with success, exit app with success

**Comment by woody34** [comment]
```
// src/completion-listener.js
...
isSuccess(exitCodes) {
        switch (this.successCondition) {
        /* eslint-disable indent */
            case 'first':
                return exitCodes[0] === 0;

            case 'last':
                return exitCodes[exitCodes.length - 1] === 0;

            case 'all-but-first':
                const [, ...allButFirst] = exitCodes;
                return allButFirst.every(code => code === 0);

            default:
                return exitCodes.every(exitCode => exitCode === 0);
            /* eslint-enable indent */
        }
    }
...
```

```
// src/completion-listener.spec.js
...
describe('with success condition set to all-but-first', () => {
    beforeEach(() => {
        commands.push(createFakeCommand('oof'));
    });

    it('succeeds if all but first process to exit has code 0', () => {
        const result = createController('all-but-first').listen(commands);

        commands[0].close.next({ exitCode: 1 });
        commands[1].close.next({ exitCode: 0 });
        commands[2].close.next({ exitCode: 0 });

        scheduler.flush();

        return expect(result).resolves.toEqual([{ exitCode: 1 }, { exitCode: 0 }, { exitCode: 0 }]);
    });

    it('fails if first process to exit has non-0 code', () => {
        const result = createController('first').listen(commands);

        commands[0].close.next({ exitCode: 1 });
        commands[1].close.next({ exitCode: 0 });
        commands[2].close.next({ exitCode: 1 });

        scheduler.flush();

        return expect(result).rejects.toEqual([{ exitCode: 1 }, { exitCode: 0 }, { exitCode: 1 }]);
    });
});
```

**Comment by gustavohenke** [comment]
hey @woody34, I'm sorry that you gave up with your PR 😢 
I put up a new PR for this, #318. It implements a more flexible/future-proof syntax for `--success` flag.

**Comment by gustavohenke** [comment]
Heya, [`v7.2.0`](https://github.com/open-cli-tools/concurrently/releases/tag/v7.2.0) is shipped with an alternative for this! 😃
